tests/protocol/mockwiredata.py

```diff
@@ -122,12 +140,19 @@ def __init__(self, data_files=DATA_FILE):
         self.ext = None
         self.remote_access = None
         self.in_vm_artifacts_profile = None
+        self.vm_artifacts_profile = None
+        self.vm_artifacts_profile_install = None
```

Comment:
This class is meant for giving more of a boiler plate structure rather than specifying specific cases inside it. A better way to achieve the same would be to do something like this - 
 
```python

    def test_ext_config_construction_dependencies(self):
        with mock_wire_protocol(mockwiredata.DATA_FILE_FAST_TRACK_DEPENDENCIES) as protocol:
            profile = InVMArtifactsProfile(protocol.mock_wire_data.in_vm_artifacts_profile)

DATA_FILE_FAST_TRACK_DEPENDENCIES = DATA_FILE.copy()
DATA_FILE_FAST_TRACK_DEPENDENCIES["ext_conf"] = "wire/ext_conf_in_vm_artifacts_profile.xml"
DATA_FILE_FAST_TRACK_DEPENDENCIES["in_vm_artifacts_profile"] = "wire/in_vm_artifacts_profile_dependencies.json"

```

You can create more `DATA_` files as per your use cases and assign them to the `in_vm_artifacts_profile` property. This makes the `WireProtocolData` more generic and usable by all external classes too and at the same time not load the extra stuff that is only used by a single test/testClass

Reply:
changed